pkgs/servers/invidious/default.nix

```diff
@@ -38,6 +38,13 @@ crystal.buildCrystalPackage rec {
       assetCommitTemplate = ''{{ "#{`git rev-list HEAD --max-count=1 --abbrev-commit -- assets`.strip}" }}'';
     in
     ''
+      # bugfix for crystal_1_12 Time::Span, can be removed when fixed upstream
+      substituteInPlace src/invidious/yt_backend/proxy.cr \
+        --replace-fail "opts[:dns_timeout] = @dns_timeout" "opts[:dns_timeout] = @dns_timeout.try(&.to_f)" \
+        --replace-fail "opts[:connect_timeout] = @connect_timeout" "opts[:connect_timeout] = @connect_timeout.try(&.to_f)" \
+        --replace-fail "opts[:read_timeout] = @read_timeout" "opts[:read_timeout] = @read_timeout.try(&.to_f)" \
```

Comment:
This is fixed in the latest release https://github.com/iv-org/invidious/releases/tag/v2.20240427 (cf. https://github.com/iv-org/invidious/pull/4570). The file `src/invidious/yt_backend/proxy.cr` was removed entirely upstream.